_scripts/import_resources.py

```diff
@@ -0,0 +1,166 @@
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+TOKEN_FILE = '.google_oauth_token'
+SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
+RESOURCES_SHEET_ID = '1QSQgxceR8BL03qsR-GY8bw4tsdsx30kovU7wVXPUcxo'
+RANGE = 'Form Responses 1!A2:J'
+
+COL_TIMESTAMP = 0
+COL_SUBMITTER = 1
+COL_MEETS_STANDARDS = 2
+COL_CATEGORY = 3
+COL_COUNTRY = 4
+COL_URL = 5
+COL_NAME = 6
+COL_DESCRIPTION = 7
+COL_STATE = 8
+COL_VERIFIED_BY = 9
+# TODO: we could write integrated_timestamp into the spreadsheet if
+# we want to have a record of this action.
+ENABLE_VERIFICATION = False
+
+
+def is_approved(row):
+    if not ENABLE_VERIFICATION:
+        return True
+    if COL_VERIFIED_BY >= len(row) or not row[COL_VERIFIED_BY]:
+        return False
+    return True
+
+
+def make_row_filename(row):
+    """Constructs standardized base filename for this resource.
+    
+    We will use YYYY-MM-DD of the submission timestamp followed by first 25
+    characters of the resource title with spaces replaced with underscores.
+    """
+   sanitized_name = row[COL_NAME].lower().replace(" ", "_")[:25]
+    dt = datetime.datetime.strptime(
+        row[COL_TIMESTAMP].split()[0],
+        '%m/%d/%Y').date()
+    return f'{dt.isoformat()}-{sanitized_name}.md'
+
+
+def sanitize_country(country):
+    """Ensure consistent country naming schemes."""
+    return {
+        "United States": "USA",
+    }.get(country, country)
+
+
+# List of dictionaries describing columns. Possible dictionary keys are:
+# - required (bool): if True, column has to be non-empty
+# - name (str): name of the column to use when emitting md files
+# - index (int): which column contains the data
+# - transform_fn (func): function that takes one argument and returns
+#     sanitized form.
+ATTRIBUTES_TO_EXTRACT = [
+    {'required': True,
+     'name': 'name',
+     'index': COL_NAME},
+    {'required': True,
+     'name': 'category',
+     'index': COL_CATEGORY},
+    {'name': 'country',
+     'index': COL_COUNTRY,
+     'transform_fn': sanitize_country},
+    {'name': 'state', 'index': COL_STATE}]
+
+
+def check_required_columns(row):
+    """Checks that all required columns are present."""
+    for attr in ATTRIBUTES_TO_EXTRACT:
+        if not attr.get('required', False):
+            continue
+        if not row[attr['index']]:
+            return False
+    return True
+
+    
+def format_row_contents(row):
+    """Generate MD output for each row."""
+    attribute_rows =[]
+    for attr in ATTRIBUTES_TO_EXTRACT:
+        col = attr['index']
+        if col >= len(row) or not row[col]:
+            continue
+        value = attr.get('transform_fn', lambda x: x)(row[col]).strip()
+        attribute_rows.append(f"{attr['name']}: {value}")
+    all_attributes = "\n".join(attribute_rows)
+    return (
+        f"---\n"
+        f"{all_attributes}\n"
+        f"\nURL: {row[COL_URL].strip()}\n"
```

Comment:
We may prefer to use some YAML library to generate the front-matter - just in case there's some escaping to be done.

Reply:
So right now, we need to move the contents into the Approved Resources column and this defacto makes this "trusted" input, but I'm happy to add sanitizer in the code. I'm not sure what's the right sanitizer to use for markdown fragments though.